packages/backend/src/index.ts

```diff
@@ -1,24 +1,58 @@
 import { createBackend } from '@backstage/backend-defaults';
+import { rootHttpRouterServiceFactory } from '@backstage/backend-defaults/rootHttpRouter';
 import { dynamicPluginsFeatureLoader } from '@backstage/backend-dynamic-feature-service';
 import { PackageRoles } from '@backstage/cli-node';
 
 import * as path from 'path';
 
 import { configureCorporateProxyAgent } from './corporate-proxy';
 import { CommonJSModuleLoader } from './loader';
-import { transports } from './logger';
+import { createStaticLogger, transports } from './logger';
 import {
   healthCheckPlugin,
   metricsPlugin,
   pluginIDProviderService,
   rbacDynamicPluginsProvider,
 } from './modules';
 
+// Create a logger to cover logging static initialization tasks
+const staticLogger = createStaticLogger({ service: 'developer-hub-init' });
+staticLogger.info('Starting Developer Hub backend');
+
 // RHIDP-2217: adds support for corporate proxy
 configureCorporateProxyAgent();
 
 const backend = createBackend();
 
+// Install a static root HttpRouterServiceFactory configure function that can
+// be explicitly overridden by the user
+if (
+  (process.env.DISABLE_STATIC_ROOT_HTTP_ROUTER_CONFIG || '').toLowerCase() !==
+  'true'
+) {
+  staticLogger.info('Using static root HttpRouterServiceFactory configuration');
+  backend.add(
+    rootHttpRouterServiceFactory({
+      configure({ app, middleware, routes }) {
```

Comment:
Do we really need this configure method, if it is simply the default service that we add back ?

Reply:
Ah, I was not aware this could be optional which would be better, let me give that a try and update this.  Thanks!